core/src/program/info.rs

```diff
@@ -0,0 +1,100 @@
+use super::{
+    ByteReader, ByteWriter, CodeBlock, Deserializable, Digest, Kernel, Program, Serializable,
+    SerializationError,
+};
+
+// PROGRAM INFO
+// ================================================================================================
+
+/// A program information set consisting of its code block root and set of kernel roots used for
+/// its compilation.
+///
+/// This will be used as public inputs of the proof so we bind its verification to the kernel and
+/// root used to execute the program. This way, we extend the correctness of the proof to the
+/// security guarantees provided by the kernel. We also allow the user to easily prove the
+/// membership of a given kernel procedure for a given proof, without compromising its
+/// zero-knowledge properties.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct ProgramInfo {
+    program_hash: Digest,
+    kernel: Kernel,
+}
+
+impl ProgramInfo {
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates a new instance of a program info.
+    pub const fn new(program_hash: Digest, kernel: Kernel) -> Self {
+        Self {
+            program_hash,
+            kernel,
+        }
+    }
+
+    /// Creates a new program info instance from a program root and a kernel set.
+    pub fn from_program_parts<R>(program_hash: R, kernel: Kernel) -> Self
+    where
+        R: AsRef<CodeBlock>,
+    {
+        Self {
+            program_hash: program_hash.as_ref().hash(),
+            kernel,
+        }
+    }
```

Comment:
Since we already implement `From` for `Program` - do we need this function?

Reply:
We don't strictly need this function. It is a way to create a `ProgramInfo` without the need to clone the entire `Program` as we don't need the CV table. With this function, we just compute the root hash and get the kernel as owned.